ow_lander/src/ow_lander/actions.py

```diff
@@ -207,6 +208,18 @@ def modify_joint_positions(self, goal):
       pos[goal.joint] = goal.angle
     return pos
 
+class ArmSetToolServer(ActionServerBase):
+
+  name          = 'ArmSetTool'
+  action_type   = owl_msgs.msg.ArmSetToolAction
+  goal_type     = owl_msgs.msg.ArmSetToolGoal
+  feedback_type = owl_msgs.msg.ArmSetToolFeedback
+  result_type   = owl_msgs.msg.ArmSetToolResult
+
+  def execute_action(self, goal):
+    msg = f"{self.name} is not yet supported in OceanWATERS."
+    rospy.logwarn(msg)
+    self._set_succeeded(msg, success=True, message=msg)
```

Comment:
The keyword args for this function should correspond to the action's Result parameters. The action Result does not have success or message, it has previous_tool.

The positional argument, which `msg` is passed to, is good. This is just whatever is logged by ActionServer when the action completes and will also be put into the Result's text parameter (this is a parameter that is internal to actionlib and exists for all actions Results).

Reply:
Thanks for pointing this out -- I wasn't familiar with the parameter scheme and just copied another call without seeing they can differ.

This did not solve the client script hang, which stumps me...